File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for p in s:
            if p == '(' or p == '{' or p == '[':
                stack.append(p)
            elif len(stack) > 0 and ((p == ')' and stack[-1] == '(') or (p == '}' and stack[-1] == '{') or (p == ']' and stack[-1] == '[')):
                stack.pop()
            else:
                return False
        return False if len(stack) > 0 else True
        

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = [];
    for (let n of s) {
        if (n === '(' || n === '{' || n === '[') {
            stack.push(n)
        } else if ((n === ')' && stack[stack.length - 1] === '(') || (n === '}' && stack[stack.length - 1] === '{') || (n === ']' && stack[stack.length - 1] === '[')) {
            stack.pop()
        } else {
            return false
        }
    }
    return stack.length > 0 ? false : true;
};
